src/main/java/me/nurio/events/EventReflection.java

```diff
@@ -49,8 +49,7 @@ public static boolean isHandledMethod(Method method) {
         }
 
         Class<?> parameter = method.getParameterTypes()[0];
-        Class<?> superClass = parameter.getSuperclass();
-        if (!superClass.equals(Event.class)) {
+        if (Event.class.isInstance(parameter.getSuperclass())) {
```

Comment:
The usage of instanceOf is not for comparing two `Class<?>` if for comparing already instantiated objects with one `Class<?>`. 

Also, you are not doing the right check, you should negate this expression to throw the exception when is not a valid instance.